Formula/sapling.rb

```diff
@@ -0,0 +1,33 @@
+class Sapling < Formula
+  desc "Source control client"
+  homepage "https://sapling-scm.com"
+  url "https://github.com/facebook/sapling/archive/refs/tags/0.1.20221118-210929-cfbb68aa.tar.gz"
+  version "0.1.20221118-210929-cfbb68aa"
```

Comment:
This looks more than a better to me.
```suggestion
  version "0.1.20221118"
```

Reply:
We decided to use both the hour (`210929`) and the commit hash (`cfbb68aa`) as part of our version number in other systems (Ubuntu, Windows). We could work around this if Homebrew has some specific guidelines against this version number.

Reply:
Are you planning on retagging the same version number multiple times that you need this additional information?

Reply:
We have [a script](https://github.com/facebook/sapling/blob/main/ci/tag-name.sh) that generates this info [every time we do a release](https://github.com/facebook/sapling/blob/main/.github/workflows/sapling-cli-windows-amd64-release.yml#L56). This script is also used for setting up the [environment variable that determines](https://github.com/facebook/sapling/blob/main/.github/workflows/sapling-cli-windows-amd64-release.yml#L24) what is shown when running `sl --version` at build time.

We had plans of using the same script for submitting PRs for updating this formula and updating this information.